src/test/enhetstester/kotlin/no/nav/familie/ba/sak/kjerne/autovedtak/svalbardtillegg/AutovedtakSvalbardtilleggBegrunnelseUtilTest.kt

```diff
@@ -0,0 +1,355 @@
+                        ytelseType = YtelseType.SVALBARDTILLEGG,
+                    ),
+                )
+
+            val innvilgedeOgReduserteSvalbardtilleggPerioder =
+                finnInnvilgedeOgReduserteSvalbardtilleggPerioder(
+                    forrigeAndeler = emptyList(),
+                    nåværendeAndeler = nåværendeAndeler,
+                )
+
+            val innvilgedePerioder = innvilgedeOgReduserteSvalbardtilleggPerioder.first
+            val redusertePerioder = innvilgedeOgReduserteSvalbardtilleggPerioder.second
+
+            assertThat(redusertePerioder).isEmpty()
+            assertThat(innvilgedePerioder).hasSize(1)
+            assertThat(innvilgedePerioder.single()).isEqualTo(YearMonth.of(2025, 11))
+        }
+
+        @Test
+        fun `Ved flere barn som både får og mister svalbardtillegg for samme periode skal det dannes innvilget og reduksjonsperiode`() {
+            val barn = lagPerson(type = PersonType.BARN)
+            val barn2 = lagPerson(type = PersonType.BARN)
+            val behandling = lagBehandling()
+            val forrigeBehandling = lagBehandling()
+
+            val forrigeAndeler =
+                listOf(
+                    lagAndelTilkjentYtelse(
+                        fom = YearMonth.of(2025, 11),
+                        tom = YearMonth.of(2025, 12),
+                        person = barn,
+                        behandling = forrigeBehandling,
+                        beløp = 500,
+                        sats = 500,
+                        ytelseType = YtelseType.SVALBARDTILLEGG,
+                    ),
+                )
+
+            val nåværendeAndeler =
+                listOf(
+                    lagAndelTilkjentYtelse(
+                        fom = YearMonth.of(2025, 11),
+                        tom = YearMonth.of(2025, 12),
+                        person = barn2,
+                        behandling = behandling,
+                        beløp = 500,
+                        sats = 500,
+                        ytelseType = YtelseType.SVALBARDTILLEGG,
+                    ),
+                )
+
+            val innvilgedeOgReduserteSvalbardtilleggPerioder =
+                finnInnvilgedeOgReduserteSvalbardtilleggPerioder(
+                    forrigeAndeler = forrigeAndeler,
+                    nåværendeAndeler = nåværendeAndeler,
+                )
+
+            val innvilgedePerioder = innvilgedeOgReduserteSvalbardtilleggPerioder.first
+            val redusertePerioder = innvilgedeOgReduserteSvalbardtilleggPerioder.second
+
+            assertThat(redusertePerioder).hasSize(1)
+            assertThat(redusertePerioder.single()).isEqualTo(YearMonth.of(2025, 11))
+            assertThat(innvilgedePerioder).hasSize(1)
+            assertThat(innvilgedePerioder.single()).isEqualTo(YearMonth.of(2025, 11))
+        }
+
+        @Test
+        fun `Ikke returner noe innvilgesesPerioder dersom andelene allerede fantes i forrige behandling`() {
```

Comment:
Det var en del mangel på l'er her 😅 Endre for finnmark også
```suggestion
        fun `Ikke returner noe innvilgelsesperioder dersom andelene allerede fantes i forrige behandling`() {
```